test-data/unit/check-optional.test

```diff
@@ -832,3 +832,99 @@ main:4: error: Argument 1 to "asdf" has incompatible type "List[str]"; expected
 main:4: note: "List" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
 main:4: note: Consider using "Sequence" instead, which is covariant
 [builtins fixtures/list.pyi]
+
```

Comment:
Other ideas for tests would be mixing different kinds of frames and control flow e.g. try/except and with statements

Reply:
Hm, related to that: maybe it's worth supporting this pattern for `try`/`except` also?

Reply:
I decided to try that, though I could be easily convinced to drop it!